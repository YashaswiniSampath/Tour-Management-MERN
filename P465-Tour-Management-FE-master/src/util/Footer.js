import React, { useState, useEffect } from 'react';
import { BottomNavigation } from '@mui/material';

const Footer = () => {
    const [isVisible, setIsVisible] = useState(false);

    const toggleVisibility = () => {
        if (window.pageYOffset + window.innerHeight >= document.documentElement.scrollHeight) {
            setIsVisible(true);
        } else {
            setIsVisible(false);
        }
    };

    useEffect(() => {
        window.addEventListener('scroll', toggleVisibility);

        return () => {
            window.removeEventListener('scroll', toggleVisibility);
        };
    }, []);

    return (
        <>
            {isVisible && (
                <BottomNavigation
                    showLabels
                    style={{
                        width: '100%',
                        position: 'fixed',
                        bottom: 0,
                        backgroundColor: '#2484BF',
                    }}
                >
                    <p style={{ color: 'white' }}>Â© 2024 EZTravel. All rights reserved.</p>
                </BottomNavigation>
            )}
        </>
    );
};

export default Footer;
