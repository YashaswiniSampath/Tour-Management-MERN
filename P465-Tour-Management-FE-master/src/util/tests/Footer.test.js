import renderer from 'react-test-renderer';
import Footer from '../Footer';

it('renders page', () => {
    const component = renderer.create(
        <Footer />
    )
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})