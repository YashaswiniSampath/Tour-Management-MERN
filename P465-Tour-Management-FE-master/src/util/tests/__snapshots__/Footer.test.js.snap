// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders page 1`] = `null`;
